src/brod_sup.erl

```diff
@@ -0,0 +1,73 @@
+%%%
+%%%   Copyright (c) 2015 Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+%%%=============================================================================
+%%% @doc
+%%% @copyright 2015 Klarna AB
+%%% @end
+%%%=============================================================================
+-module(brod_sup).
+-behaviour(brod_supervisor).
+
+-export([ start_link/1
+        , init/1
+        ]).
+
+-define(DEFAULT_PRODUCER_RESTART_DELAY_SECONDS, 10).
+
+%% @doc Start supervisor of permanent producers.
+%% @see brod:start_link_producer/2 for more info about all producer args
+%% @end
+-spec start_link([proplists:proplist()]) -> {ok, pid()}.
+start_link(PermanentProducers) ->
+  brod_supervisor:start_link({local, ?MODULE}, ?MODULE, PermanentProducers).
+
+init(PermanentProducers) ->
+  Children = [permanent_producer(P) || P <- PermanentProducers],
```

Comment:
To interact with the producers you need to have their pids. We need an API to get the pids of the permanent producers from the supervisor.